task_processing/plugins/kubernetes/task_config.py

```diff
@@ -0,0 +1,52 @@
+import uuid
+from typing import Sequence
+from typing import TYPE_CHECKING
+
+from pyrsistent import field
+from pyrsistent import m
+from pyrsistent import PMap
+from pyrsistent import pmap
+from pyrsistent import PVector
+from pyrsistent import pvector
+from pyrsistent import v
+
+from task_processing.interfaces.task_executor import DefaultTaskConfigInterface
+
+VOLUME_KEYS = set(['mode', 'container_path', 'host_path'])
+
+def valid_volumes(volumes):
+    for vol in volumes:
+        if set(vol.keys()) != VOLUME_KEYS:
+            return (
+                False,
+                'Invalid volume format, must only contain following keys: '
+                '{}, was: {}'.format(VOLUME_KEYS, vol.keys())
+            )
+    return (True, None)
+
+class KubernetesTaskConfig(DefaultTaskConfigInterface):
+    uuid = field(type=(str, uuid.UUID), initial=uuid.uuid4)
+    node_name = field(type=str)
+    node_selector = field(type=(str,str))
+    containers = field(type=PVector)
+    restart_policy = field(type=str)
```

Comment:
hmm, can we do an enum here? there's a finite set of restart policies, so it'd be nice to have a more ergonomic way of setting a restart policy than just passing in arbitrary strings (since they're easy to typo)

I think we can also set a default of `Never` since I think we can assume that task_proc users (at least internally) want to signal that their tasks are idempotent by explicitly setting a restart policy themselves - and both Tron (and Jolt, iirc) handle retries themselves anyway so it'd be nicer for us too :p

Reply:
or actually, maybe just hardcode to Never for now? I think it's not worth implementing any retry/restart stuff in task_proc just yet since our initial user (Tron) won't use it - I think Jolt does, but we can cross that bridge when we get there since it's not yet clear what Jolt on k8s will look like

Reply:
Done.